test/metabase/segments/api_test.clj

```diff
@@ -68,33 +85,40 @@
-    (is (= {:name                    "A Segment"
-            :description             "I did it!"
-            :show_in_getting_started false
-            :caveats                 nil
-            :points_of_interest      nil
-            :creator_id              (mt/user->id :crowberto)
-            :creator                 (user-details (mt/fetch-user :crowberto))
-            :entity_id               true
-            :created_at              true
-            :updated_at              true
-            :archived                false
-            :definition              {:filter ["=" ["field" 10 nil] 20]}}
-           (segment-response (mt/user-http-request :crowberto :post 200 "segment"
-                                                   {:name                    "A Segment"
-                                                    :description             "I did it!"
-                                                    :show_in_getting_started false
-                                                    :caveats                 nil
-                                                    :points_of_interest      nil
-                                                    :table_id                id
-                                                    :definition              {:filter [:= [:field 10 nil] 20]}}))))))
+    (doseq [[format-name definition-fn] {"MBQL4" mbql4-segment-definition
+                                         "pMBQL" pmbql-segment-definition}]
+      (testing format-name
+        (is (= {:name                    "A Segment"
+                :description             "I did it!"
+                :show_in_getting_started false
+                :caveats                 nil
+                :points_of_interest      nil
+                :creator_id              (mt/user->id :crowberto)
+                :creator                 (user-details (mt/fetch-user :crowberto))
+                :entity_id               true
+                :created_at              true
+                :updated_at              true
+                :archived                false
+                :definition true}
+               (-> (mt/user-http-request :crowberto :post 200 "segment"
+                                         {:name                    "A Segment"
+                                          :description             "I did it!"
+                                          :show_in_getting_started false
+                                          :caveats                 nil
+                                          :points_of_interest      nil
+                                          :table_id                id
+                                          :definition (if (= format-name "pMBQL")
+                                                        (definition-fn id 10 20)
+                                                        (definition-fn 10 20))})
+                   segment-response
+                   (update :definition map?))))))))
 
```

Comment:
Maybe I missed a memo, but I thought our convention was that one could choose not to align columns, but should preserve such alignment where it already exists. Maybe we never really reached a conclusion on that one.